SimDataFormats/TrackerDigiSimLink/interface/PixelSimHitAddExtraInfo.h

```diff
@@ -0,0 +1,41 @@
+#ifndef PixelSimHitAddExtraInfo_h
+#define PixelSimHitAddExtraInfo_h
+
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include <vector>
+#include <cstdint>
+
+class PixelSimHitAddExtraInfo {
+public:
+  PixelSimHitAddExtraInfo(size_t Hindex, Local3DPoint entryP, Local3DPoint exitP, unsigned int ch) {
+    index_ = Hindex;
+    theEntryPoint_ = entryP;
+    theExitPoint_ = exitP;
+    chan_.push_back(ch);
+  };
+  PixelSimHitAddExtraInfo(){};
+  ~PixelSimHitAddExtraInfo(){};
+  size_t hitIndex() const { return index_; };
+  Local3DPoint entryPoint() const { return theEntryPoint_; };
```

Comment:
Again, for a more efficient code, please define all getter functions as const references, e.g. `const Local3DPoint& entryPoint() const { return theEntryPoint_; };`